From: Matthew Wilcox <willy@infradead.org>
Subject: khugepaged-use-a-folio-throughout-collapse_file-fix
Date: Sun, 7 Apr 2024 04:43:27 +0100

the unlikely() is embedded in IS_ERR() so no need to keep it here

Link: https://lkml.kernel.org/r/ZhIWX8K0E2tSyMSr@casper.infradead.org
Signed-off-by: Matthew Wilcox <willy@infradead.org>
Reported-by: Vishal Moola <vishal.moola@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/khugepaged.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mm/khugepaged.c~khugepaged-use-a-folio-throughout-collapse_file-fix
+++ a/mm/khugepaged.c
@@ -1865,7 +1865,7 @@ static int collapse_file(struct mm_struc
 				/* drain lru cache to help isolate_lru_page() */
 				lru_add_drain();
 				folio = filemap_lock_folio(mapping, index);
-				if (unlikely(folio == NULL)) {
+				if (IS_ERR(folio)) {
 					result = SCAN_FAIL;
 					goto xa_unlocked;
 				}
_
